inputfiles/addedTypes.jsonc

```diff
@@ -1374,6 +1463,21 @@
             {
                 "name": "EventListenerOrEventListenerObject",
                 "overrideType": "EventListener | EventListenerObject"
+            },
+            {
+                "name": "AutoFillSection",
+                "overrideType": "`section-${string}`"
```

Comment:
This is problematic as it allows space: `section-foo bar baz` is allowed as `AutoFillSection`.

Reply:
Honestly it doesn't seem trivial, I don't think I'm fluent enough in typescript to do this (or if it's possible at all).

I will add tests that show the expected and unexpected cases anyway.

I'll continue looking into it but I would argue that the current type (before this PR) allows much more invalid values, so maybe we could iterate on it? Or do you feel like it would be a blocker?

Reply:
I can see how to do it with a generic, but not inside a template literal type:

[playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggrsA9gMQJYBt0DlECcC2AhumhOgCZQC8UARIeeasKogHbEC0H+EtUAHzoNyuCAGdxndBABuZAIz8h9RmMnS5ZAEzLhaiVJnz0AZj2rRhzSYAsFkeqOo2EJYP1WN6FxF0fLJ2lfcwCAI3JCEE5IkAsIqM58dmAAC3jYzhAIQlwLAGN8znzxfIKiiAAPMHLOKrAklPSAwrrqrJy8lqKAM0J8DGiePm7OAHNUeTZufpGVVuHatjh8MIgu+aLQSALEODZgXDiWvYOjmd4CuFwxA84wQkkAdzxyCz6B9CHZiwmpi7mdFS7DwqB6qCKYDE4MqFmBbFB4KK4jgPRhFkWAVcT3uj3EL1wbwC7AgnBYvGKiHIgNoeDGhDYqAAXoQWOwLGBEOJgFx8lSadyxBBgJxHIYLIcGeJCPk2dN+qdgBLcFKZXLitdbvljio4OJ1osANwAWAAUNtoPAkGhMABhFJqgDSLkoNFowMuAWSYQwNIJAGtaCbzeBLQgUBh0PaDmrSBRqHQIEQMBKyKn0KKxIRKdT05T9odonzcwFgGQ2mW2OJWGw8+hEPliHWG1woRB0aXy-XGxmUWjULDOxmOHKm8GLbBwzb0DADJJnWxXXQfZgXGMLOJUqgwGA10GzROrRG7WJqQdUMQ40vaE8IGFCAhUrXg2azQB6N9QTjf1+mg+hqAcAAZQefIICAiBZRrAAeAAVKAqkrchxCgQU1wAPgTeDEIgRcUIAA31KD2E4AASABvFwenWKAACUJGAABffCzSgNioAAfjohiEMqJCCIotC2DGRioEEw412Y1j2K41x5FwaS2IALigWDFKgFS5PWF9TVkXIoEIFTgNA8DILlaCAHIiPVHpEEQe9cAszCaCssya04Wz7NyCzgw-KAAAFgCkeozLqG48DNPTcCgMIjMQECZVM4i2Es6z3M8mLvOcqBXOSjy7MyxydMPKdIwg5KE2MxLyvM-DxNwST8PQnSoqgfIVKPacaprBNcpsuyHJ8s18nfT9AuC6pQvWXAIt0-TyA60rMG69herSkiMsG4NyF-Pzv04X8JwAeTAUd0AABWhAdYMQf1cLg3j+NQiThOyuryNg0T8ICWh-0gKATrO86uWAGEbrulLsL43DkOehrXoTb6KM+76VF+v9TT8gAKTrIxwAhL1QMhKCEHGlqjB1ZQXcguNxu1KeAK8AEomb+sNrUjK8Ezp7A8CIEgifjIR3sBmtiEu9trtu+6eejHkqZddDGIo2WGavZidL8o9wWWtz2Fp8MdZnOdxGpg2OcwLnZdPXCWEJ4mOLZycLfQRGKNF9hxauypwfuqqwJWlKecD9Clfd06xYu73faD8nZy8U3FeV8iefV8OgZBsHpdjl3bRt897YoJWWIx0avx-UvMbGoK2kgWUwpmhS5uiuBFpdta9emTb9L1A1ZiG004Ei-SQDb49XZc9au4K3vcGGCyzRAX9WsQMfp162f57L8ba6m8Km9asA18jDf9Wi+egA)

Reply:
Yes, pretty sure it's not currently possible. Anyway as I said, still much less false positives than before. :)

Reply:
>Anyway as I said, still much less false positives than before. :)

While I admit, it may give false impression that any match would be legal when it's not, which may be more harmful.

```ts
// Has typos but no error, because AutoFillSection greedily matches the typos :(
var illegal: AutoFill = 'section-typescript billin hom tel-local'
```

Perhaps it's okay to have some helper generics? I want to hear from Microsoft people.

Reply:
@sandersn This one is the main reason I'm not approving, what do you think?

Reply:
I think some overgeneration is OK as long as it's not rejecting values that are legal.
I do wonder about performance though. Let me look again at the generated size of the type.

Reply:
> This is problematic as it allows space: `section-foo bar baz` is allowed as `AutoFillSection`.

Note for other readers that it is not a full catch-all, you still need to add valid tokens to your attribute, so for example, `section-foo bar baz username` would pass but not `section-foo bar baz` alone, nor `section-wrong off` as shown in the tests.

Reply:
> I do wonder about performance though. Let me look again at the generated size of the type.

@sandersn: Hi, I've pushed the final version, are you ok with the generated types?

https://github.com/microsoft/TypeScript-DOM-lib-generator/pull/1467/files#diff-d57c51d024219b9f37e6b42df127db25f700acbaade23ce1edcecda2853b995c

Reply:
Can you add a TODO comment about the over-match of `section-{string}`?